src/main/kotlin/org/jetbrains/research/libsl/nodes/variables.kt

```diff
@@ -40,21 +50,62 @@ open class Variable(
 
 data class ResultVariable(
     override var typeReference: TypeReference
-) : Variable(name = "result", typeReference)
+) : Variable(keyword = VariableKeyword.RESULT, name = "result", typeReference)
 
 @Suppress("unused")
 class FunctionArgument(
     name: String,
     typeReference: TypeReference,
     val index: Int,
-    var annotation: Annotation? = null,
+    var annotationsReferences: MutableList<AnnotationReference>? = mutableListOf(),
     var targetAutomaton: AutomatonReference? = null
-) : Variable(name, typeReference) {
+) : Variable(keyword = VariableKeyword.ARGUMENT, name, typeReference) {
     lateinit var function: Function
 
     override val fullName: String
         get() = "${function.name}.$name"
 
+    override fun dumpToString(): String = buildString {
+        if (annotationsReferences != null) {
+
+            annotationsReferences?.joinToString() { annotation ->
+                val resolvedAnnotation = annotation.resolveOrError()
+                append("@")
+                append(BackticksPolitics.forIdentifier(resolvedAnnotation.name))
+
+                if (resolvedAnnotation.values.isNotEmpty()) {
+                    append("(")
+                    append(resolvedAnnotation.values.joinToString(separator = ", ") { v ->
+                        v.dumpToString()
+                    })
+                    append(")")
+                }
+
+                append(IPrinter.SPACE)
+            }
+        }
+        append(BackticksPolitics.forIdentifier(name))
+        append(": ")
+
+        val typeName = if (targetAutomaton != null) {
+            targetAutomaton!!.name
+        } else {
+            typeReference.name
+        }
+
+        append(typeName)
```

Comment:
As far as I can remember, not always argument with `@target` annotation has the same type as an automaton's type passed to it via such kind of annotations